integration/gitopscreateprofile_test.go

```diff
@@ -0,0 +1,90 @@
+// +build integration
+
+package integration_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gexec"
+	"github.com/spf13/afero"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+var _ = Describe("(Integration) generate profile", func() {
+
+	testDirectory := "test_profile"
+
+	BeforeSuite(func() {
+		kubeconfigTemp = false
+		if kubeconfigPath == "" {
+			wd, _ := os.Getwd()
+			f, _ := ioutil.TempFile(wd, "kubeconfig-")
+			kubeconfigPath = f.Name()
+			kubeconfigTemp = true
+		}
+	})
+
+	AfterSuite(func() {
+		gexec.KillAndWait()
+		os.RemoveAll(testDirectory)
+	})
+
+	Describe("when generating a profile", func() {
+		It("should write the processed repo files in the supplied directory", func() {
+
+			clusterName = "amazing-testing-gopher"
+
+			eksctlSuccess("generate", "profile",
+				"--verbose", "4",
+				"--name", clusterName,
+				"--region", region,
+				"--git-url", "git@github.com:eksctl-bot/eksctl-profile-integration-tests.git",
+				"--profile-path", testDirectory,
+			)
+
+			fs := afero.Afero{
+				Fs: afero.NewOsFs(),
+			}
+
+			contents, err := fs.ReadFile(filepath.Join(testDirectory, "workloads/namespace.yaml"))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(string(contents)).To(Equal(
+				`apiVersion: v1
+kind: Namespace
+metadata:
+  labels:
+    name: amazing-testing-gopher
+  name: amazing-testing-gopher
+`))
+
+			contents, err = fs.ReadFile(filepath.Join(testDirectory, "workloads/services/service.yaml"))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(string(contents)).To(Equal(
+				`apiVersion: v1
```

Comment:
I must admit, I do find inline YAML very irritating as it messes with indentation. But I can tolerate it in tests.

Declaring it with `---` on the first line does make it a little more readable.

Reply:
I agree, Is there a better way? I can for example move it to a constant somewhere else, would that be better?

Reply:
A constant at the bottom of the file would be nice :)

Reply:
Actually, for tests we could just have a file in 'testdata/', but these are fairly short snippets, so it is kind of more convenient to have theme inline...